Toolchain paths fix (linux)

kerberjg opened the pull request:
Fixes some toolchain issues on workspace init:

- On Fedora, LLVM would ignore `PREFER_LLVM` and always install latest OS default instead
- On Linux, `cmake` and `ninja` were not included in the `setup_env.sh` path, causing them to work during the initial build but never after

Technically this should also remove the requirement of having CMake/Ninja as pre-workspace dependencies

kerberjg committed:
Fixed LLVM not installing the right version on Fedora

kerberjg committed:
Fixed non-LLVM toolchains not having their paths included

jwinarske commented on flutter_workspace.py:
```diff
@@ -2139,20 +2139,21 @@ def setup_toolchain(platform_, git_token, cookie_file, plex, enable, disable, en
-                for line in append_to_runtime_env:
-                    append_to_env_script(workspace, line)
-
     elif platform_['toolchain'] == 'common':
```
```
    elif platform_['toolchain'] == 'common':
        pass
    else:
        print_banner("Toolchain not supported")

    # append lines to runtime env script (for both llvm and common toolchains)
    workspace = os.environ.get('FLUTTER_WORKSPACE')
    if 'append_to_runtime_env' in platform_:
        append_to_runtime_env = platform_['append_to_runtime_env']
        if append_to_runtime_env:
            append_to_env_script(workspace, '\n')
            for line in append_to_runtime_env:
                append_to_env_script(workspace, line)
```

jwinarske commented:
the checks need to pass on all targets.

jwinarske commented on configs/toolchain-common-cmake.json:
```diff
@@ -18,6 +18,10 @@
     },
+    "append_to_runtime_env": [
+        "# toolchain-common-cmake",
+        "export PATH=$FLUTTER_WORKSPACE/.config/flutter_workspace/toolchain-common-cmake/bin:$PATH"
```
`$PLATFORM_ID_DIR` should be `$FLUTTER_WORKSPACE/.config/flutter_workspace/toolchain-common-cmake`

jwinarske commented on configs/toolchain-common-ninja.json:
```diff
@@ -19,6 +19,10 @@
     },
+    "append_to_runtime_env": [
+        "# toolchain-common-cmake",
+        "export PATH=$FLUTTER_WORKSPACE/.config/flutter_workspace/toolchain-common-ninja:$PATH"
```
$PLATFORM_ID_DIR should be $FLUTTER_WORKSPACE/.config/flutter_workspace/toolchain-common-ninja

jwinarske committed:
toolchain common env append cleanup

-use pass instead of an empty print
-move append after common check

Signed-off-by: Joel Winarske <joel.winarske@gmail.com>

jwinarske commented:
@kerberjg this fails on Windows, which requires different handling.

Error
```
********************************************
* Setting FLUTTER_WORKSPACE to:
* D:\a\workspace-automation\workspace-automation
********************************************
export: D:\a\workspace-automation\workspace-automation\setup_env.ps1:26
Line |
  26 |  export PATH=$FLUTTER_WORKSPACE/.config/flutter_workspace/toolchain-co …
     |  ~~~~~~
     | The term 'export' is not recognized as a name of a cmdlet, function, script file, or executable program. Check
     | the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

`append_to_runtime_env` requires a windows key, so then if host is windows, you can use the correct key/value pair for toolchain/common.

Your initial use of the skip windows label masked this.

jwinarske committed:
append_to_runtime_env update

-checks for host_type/distro
 if host_type/distro present, uses that
 if common is present, uses that
 else will not append anything
 This allows windows to override "common" host type.
 "common" host type is a virtual construct.

Signed-off-by: Joel Winarske <joel.winarske@gmail.com>
